### DISCUSSION TITLE
List, Create, and Update Members via EECli Tool

### DISCUSSION THREAD

**Discussion by jaredchesebro** [ExpressionEngine/ExpressionEngine#2892] [Feature Requests]
## Problems Solved
I think it solves the following issues:
- a developer leaves a team and he was the only super admin
- when the forgot password email isn't received and you need to change a password

## Description
The ability to create a new user or update the following existing user credentials via the eecli tool.
- email
- username
- password
- group

It would also be great to be able to list the existing users so you can reference them in the update command.

### Commands

I'd expect the commands to be something like this:

**List Members**
```php system/eecli.php members:list --group={groupname}```

**Create New Member**
```php system/eecli.php members:create --username={username} --email={email} --password={password} --group={group}```

**Update Existing Member**
```php system/eecli.php members:update --username={newusername} --email={newemail} --password={newpassword} --group={newgroup}```

_*replace {variables} with applicable values_

**Comment by jaredchesebro** [comment]
@intoeetive Thanks, Yuri.

**Comment by eric-lamb62** [comment]
@jaredchesebro this is very interesting. I worry about the precedent, but it's an itchy request. I kinda like the idea tbh, 

I can just so very easily see how this could escalate into the same for Channel Entries, and Roles, and Sites, and on (not that those are bad ideas, it's just we'd be adding another input layer). So would be smart to start (lol) with an execution abstraction so there's uniformity with the direction(s). 

@intoeetive was this something you guys internally are wanting to pursue? I'm down to scratch this, if so.

**Comment by intoeetive** [comment]
@matthewjohns0n would know better on the CLI front here

**Comment by matthewjohns0n** [comment]
We havent discussed the member side of things, but we have definitely talked about managing other things via the CLI like channels, fields, etc.

I definitely understand the desire for this, and if you have CLI access you are essentially already a super admin, but it gives enough access that I would want to discuss internally whether this would be desired for a core feature.

**Comment by eric-lamb62** [comment]
This may be the worst idea ever, but Tinker with/for EE Model objects?
https://github.com/laravel/tinker

**Comment by eric-lamb62** [comment]
But, I mean, native.
